app/src/main/java/com/example/fundatecheroes2/login/view/LoginActivity.kt

```diff
@@ -4,20 +4,34 @@ import android.content.Intent
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
 import android.widget.Toast
+import androidx.activity.viewModels
+import com.example.fundatecheroes.login.view.presentation.LoginViewModel
+import com.example.fundatecheroes.login.view.presentation.ViewState
 import com.example.fundatecheroes2.databinding.ActivityLoginBinding
 import com.example.fundatecheroes2.home.HomeActivity
 import com.example.fundatecheroes2.profile.view.ProfileActivity
+import com.fundatec.components.test
 
 class LoginActivity : AppCompatActivity() {
 
     private lateinit var binding: ActivityLoginBinding
 
+    private val viewModel: LoginViewModel by viewModels()
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityLoginBinding.inflate(layoutInflater)
         setContentView(binding.root)
 
         configLoginButton()
+        viewModel.viewState.observe(this) { state ->
+            when (state) {
+                is
```

Comment:
A principio não precisamos desse `is` para esses cenários, precisariamos dele caso algum dos ViewStates fosse uma class